### PR TITLE
Use disable_sigint instead of sigatomic_(begin|end)

### LINKED ISSUE #684: Use disable_sigint instead of sigatomic_(begin|end)
`disable_sigint` is the public API so we better use it instead of `sigatomic_(begin|end)`.

A side effect is that it makes PyCall work with JuliaInterpreter.jl ouf-of-the-box and avoid issues like https://discourse.julialang.org/t/23560 and https://github.com/JuliaDebug/JuliaInterpreter.jl/issues/219.  This is because JuliaInterpreter.jl uses `disable_sigint` to workaround the issues related to the signal handler https://github.com/JuliaDebug/JuliaInterpreter.jl/pull/108.

@stevengj Repeating what I asked in #683:

> @stevengj You didn't want to put `disable_sigint` for _all_ `ccall` https://github.com/JuliaPy/PyCall.jl/pull/574#discussion_r219716154 but maybe replacing only the `sigatomic_begin`-`sigatomic_end` pairs is OK?

### PR DESCRIPTION
**Pull request by tkf**
close #684

### CODE REVIEW

**Comment by tkf** [comment]
So it seems that just switching to `disable_sigint` impedes Julia to inline `ccall`s.  But we can avoid it with the optimization 6ca05f4591113a9950a500aba1c1552c071bfb6c mentioned in https://github.com/JuliaLang/julia/issues/29688.

Before 6ca05f4591113a9950a500aba1c1552c071bfb6c

```
julia> code_warntype(PyCall.__pycall!, (PyObject, PyPtr, PyObject, Ptr{Cvoid}); optimize=true)
Variables
  #self#::Core.Compiler.Const(PyCall.__pycall!, false)
  ret::PyObject
  pyargsptr::Ptr{PyCall.PyObject_struct}
  o::PyObject
  kw::Ptr{Nothing}
  #110::getfield(PyCall, Symbol("##110#111")){PyObject,Ptr{PyCall.PyObject_struct},PyObject,Ptr{Nothing}}

Body::PyObject
1 ─ %1 = %new(getfield(PyCall, Symbol("##110#111")){PyObject,Ptr{PyCall.PyObject_struct},PyObject,Ptr{Nothing}}, ret, pyargsptr, o, kw)::getfield(PyCall, Symbol("##110#111")){PyObject,Ptr{PyCall.PyObject_struct},PyObject,Ptr{Nothing}}
│        invoke PyCall.disable_sigint(%1::getfield(PyCall, Symbol("##110#111")){PyObject,Ptr{PyCall.PyObject_struct},PyObject,Ptr{Nothing}})
└──      return ret
```

After 6ca05f4591113a9950a500aba1c1552c071bfb6c

```
julia> code_warntype(PyCall.__pycall!, (PyObject, PyPtr, PyObject, Ptr{Cvoid}); optimize=true)
Variables
  #self#::Core.Compiler.Const(PyCall.__pycall!, false)
  ret::PyObject
  pyargsptr::Ptr{PyCall.PyObject_struct}
  o::PyObject
  kw::Ptr{Nothing}
  retptr::Ptr{PyCall.PyObject_struct}
  val::Ptr{PyCall.PyObject_struct}

Body::PyObject
1 ─       $(Expr(:foreigncall, :(:jl_sigatomic_begin), Nothing, svec(), :(:ccall), 0))
│   %2  = PyCall.getfield(o, :o)::Ptr{PyCall.PyObject_struct}
│   %3  = Base.bitcast(Ptr{PyCall.PyObject_struct}, kw)::Ptr{PyCall.PyObject_struct}
│   %4  = $(Expr(:foreigncall, :(Core.tuple(:PyObject_Call, PyCall.libpython)), Ptr{PyCall.PyObject_struct}, svec(Ptr{PyCall.PyObject_struct}, Ptr{PyCall.PyObject_struct}, Ptr{PyCall.PyObject_struct}), :(:ccall), 3, :(%2), :(pyargsptr), :(%3), :(kw), :(pyargsptr), :(o)))::Ptr{PyCall.PyObject_struct}
│   %5  = Core.bitcast(Core.UInt, %4)::UInt64
│   %6  = (%5 === 0x0000000000000000)::Bool
└──       goto #7 if not %6
2 ─ %8  = $(Expr(:foreigncall, :(Core.tuple(:PyErr_Occurred, PyCall.libpython)), Ptr{PyCall.PyObject_struct}, svec(), :(:ccall), 0))::Ptr{PyCall.PyObject_struct}
│   %9  = Core.bitcast(Core.UInt, %8)::UInt64
│   %10 = (%9 === 0x0000000000000000)::Bool
│   %11 = Base.not_int(%10)::Bool
└──       goto #4 if not %11
3 ─ %13 = invoke PyCall.PyError("\$(Expr(:escape, :(ccall(#= /home/takafumi/.julia/dev/PyCall/src/pyfncall.jl:43 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw))))"::String)::PyCall.PyError
│         PyCall.throw(%13)
└──       $(Expr(:unreachable))
4 ┄       goto #5
5 ─       goto #6
6 ─       invoke PyCall.error("\$(Expr(:escape, :(ccall(#= /home/takafumi/.julia/dev/PyCall/src/pyfncall.jl:43 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw))))"::String, " failed"::String)
└──       $(Expr(:unreachable))
7 ┄       invoke PyCall.pydecref_(_2::PyObject)
│         PyCall.setfield!(ret, :o, %4)
│         $(Expr(:foreigncall, :(:jl_sigatomic_end), Nothing, svec(), :(:ccall), 0))
└──       return ret
```

**Comment by stevengj** [comment]
What is the relationship of this PR to #574?

**Comment by tkf** [comment]
The first commit c030fdd72d347ffcb47615aec4e551489a58d719 in this PR is included in #574 but #574 comes with more "destructive" changes.  This is a safe subset of #574.

The second commit 6ca05f4591113a9950a500aba1c1552c071bfb6c is a fix for the possible performance regression (`ccall` not inlined) due to the first commit.  I thought this might be appropriate given your comment https://github.com/JuliaPy/PyCall.jl/pull/574#discussion_r219716154.

File: src/pyeval.jl [hunk 1]
```diff
@@ -31,16 +31,13 @@ pynamespace(m::Module) =
 # and a current "file name" to use for stack traces
 function pyeval_(s::AbstractString, globals=pynamespace(Main), locals=pynamespace(Main),
                  input_type=Py_eval_input, fname="PyCall")
-    sigatomic_begin()
-    try
+    disable_sigint() do
```
Comment (stevengj) [inline src/pyeval.jl]: Shouldn't this be just before the `PyEval_EvalCode` line?

Comment (tkf) [reply src/pyeval.jl]: I pulled out `Py_CompileString` from `disable_sigint` in d5ee7551778d3256b9dfc20a9a4b8c2494882551.  See the commit message.

**Comment by tkf** [comment]
AppVeyor failures are the same as #687.  Merging...

I'm not using squash since each commit has distinct information.

**Pull request merged**